File: openclaw_safe/channels/telegram.py
"""Telegram channel adapter."""

from __future__ import annotations

from typing import Any

from .base import BaseChannel


class TelegramChannel(BaseChannel):
    """Adapter for the Telegram Bot API."""

    channel_name = "telegram"

    def __init__(self, bot_token: str) -> None:
        self._bot_token = bot_token

    def send_message(self, recipient_id: str, text: str, **kwargs: Any) -> None:
        # In production: POST to https://api.telegram.org/bot<token>/sendMessage
        raise NotImplementedError(
            "TelegramChannel.send_message requires a live bot token and HTTP client."
        )

    def receive_message(self, payload: dict[str, Any]) -> dict[str, Any]:
        update = payload.get("message", payload)
        return {
            "sender_id": str(update.get("from", {}).get("id", "")),
            "text": update.get("text", ""),
            "channel": self.channel_name,
            "raw": payload,
        }

File: src/channels/telegram.ts
import { BaseChannel, NormalisedMessage } from "./base";

export class TelegramChannel extends BaseChannel {
  readonly channelName = "telegram";

  constructor(private readonly botToken: string) {
    super();
  }

  async sendMessage(_recipientId: string, _text: string): Promise<void> {
    throw new Error(
      "TelegramChannel.sendMessage requires a live bot token and HTTP client."
    );
  }

  receiveMessage(payload: unknown): NormalisedMessage {
    const p = payload as Record<string, unknown>;
    const msg = (p["message"] ?? p) as Record<string, unknown>;
    const from = (msg["from"] ?? {}) as Record<string, unknown>;
    return {
      senderId: String(from["id"] ?? ""),
      text: String(msg["text"] ?? ""),
      channel: this.channelName,
      raw: payload,
    };
  }
}
